glusterd2/transaction/rpc-service.go

```diff
@@ -19,35 +19,44 @@ func init() {
 
 // RunStep handles the incoming request. It executes the requested step and returns the results
 func (p *txnSvc) RunStep(rpcCtx context.Context, req *TxnStepReq) (*TxnStepResp, error) {
-	var ctx Tctx
 
-	err := json.Unmarshal(req.Context, &ctx)
-	if err != nil {
+	var (
+		resp   TxnStepResp
+		f      StepFunc
+		err    error
+		ok     bool
+		logger log.FieldLogger
+	)
+
+	var ctx Tctx
+	if err = json.Unmarshal(req.Context, &ctx); err != nil {
```

Comment:
Instead of embedding the context marshalled to JSON, we can now just define the context config as a protobuf message struct and add it to the TxnStepReq. The only reason I went with embedding json was because I didn't find a way to implement custom protobuf marshalling for the old-style contexts with parent contexts.

Reply:
The field logger only has txnid as the only field. Hence, would it suffice is only txn id is sent on wire ?

Reply:
> The field logger only has txnid as the only field. Hence, would it suffice if only txn id is sent on wire ?

This sounds okay with me. The only reason the full TxnCtx was being on-wire was because the first implementation didn't use the store to save txn data. So I had send across the full context. Now that we use the store, it's enough that we just pass along the identifier. The remote end can get the context from the store.

Reply:
> The type logrus.Fields is of type map[string]interface{} and there's no simple way to define that in proto syntax. Proto does support map but the value must be a concrete type. A simpler alternative is to use gob or json which supports map[string]interface{}.

Oh well. That was another reason why I decided to embed JSON in protobuf.